Skynet/src/main/java/Factory.java

```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class Factory {
```

Comment:
Race Conditions: The factions access and collect parts from the factory's queue concurrently. Since there's no synchronization, two factions could try to take the same part, or more likely, one faction could collect its parts while another is also collecting, leading to an unpredictable division of parts. A ConcurrentLinkedQueue prevents data corruption, but it doesn't solve the race for resources.

The produceParts method: The factory's produceParts method calls availableParts.clear(). This is a huge problem. It means that while factions are trying to collect parts, the factory can suddenly wipe the queue clean, causing factions to collect nothing even if parts were just produced. The factory and factions need to work in lock-step, not in a chaotic, unsynchronized manner.